pkg/networkmanager/interfaces.js

```diff
@@ -785,18 +824,25 @@ export function NetworkManagerModel() {
             connections_by_uuid[settings.connection.uuid] = obj;
     }
 
-    function refresh_settings(obj) {
+    async function refresh_settings(obj) {
         push_refresh();
-        client.call(objpath(obj), "org.freedesktop.NetworkManager.Settings.Connection", "GetSettings")
-                .then(function(reply) {
-                    const result = reply[0];
-                    if (result) {
-                        priv(obj).orig = result;
-                        set_settings(obj, settings_from_nm(result));
-                    }
-                })
-                .catch(complain)
-                .finally(pop_refresh);
+
+        try {
+            const reply = await client.call(objpath(obj), "org.freedesktop.NetworkManager.Settings.Connection", "GetSettings");
+            const result = reply[0];
+            if (result) {
+                if (result.wireguard) {
+                    const secrets = await client.call(objpath(obj), "org.freedesktop.NetworkManager.Settings.Connection", "GetSecrets", ["wireguard"]);
+                    result.wireguard['private-key'] = secrets[0].wireguard['private-key'];
```

Comment:
Same here, write it as `const [secrets] = await ...` to avoid the `[0]`. D-Bus calls always return a list.